WSLGd/FontMonitor.cpp

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+#include "FontMonitor.h"
+#include "common.h"
+
+#define USER_DISTRO_MOUNT_PATH "/mnt/wslg/distro"
+#define USER_DISTRO_FONTPATH USER_DISTRO_MOUNT_PATH "/usr/share/fonts/"
+
+constexpr auto c_fontsdir = "fonts.dir";
+constexpr auto c_xset = "/usr/bin/xset";
+
+wslgd::FontFolder::FontFolder(int fd, const char *path)
+{
+    LOG_INFO("FontMonitor: start monitoring %s", path);
+
+    m_fd.reset(fd);
+    m_path = path;
+
+    /* check if folder is already ready to be added to font path */
+    try {
+        std::string fonts_dir(path);

```

Comment:
This seems to rely on path ending with a '/', would probably be better to use std::filesystem::path here.

Reply:
I personally feel their usage of "/=" operator seems more confusing, thus avoided that, but I will switch to that, thanks.